core/src/mender-artifact.c

```diff
@@ -358,9 +363,12 @@ mender_artifact_process_data(mender_artifact_ctx_t *ctx, void *input_data, size_
             new_size = ctx->input.length + input_length;
             /* data/ files are processed per block for which the original size of the buffer should
                be enough, but metadata is processed as whole files so there we expect we will need
-               more. */
+               more, except for header.tar (and tarballs in general) which are processed
+               transparently. */
             if (mender_utils_strbeginswith(ctx->file.name, "data/")) {
                 expected_required = ctx->input.orig_size;
+            } else if (mender_utils_strendswith(ctx->file.name, ".tar")) {
+                expected_required = ctx->input.orig_size;
```

Comment:
I think it would read cleaner: 

```suggestion
            if ((mender_utils_strbeginswith(ctx->file.name, "data/")) ||  (mender_utils_strendswith(ctx->file.name, ".tar"))) {
                expected_required = ctx->input.orig_size;
```

Reply:
Just by the look at the diffs I don't agree, but YMMV.